### DISCUSSION TITLE
A/B test on first load

### DISCUSSION THREAD

**Discussion by etticat** [adinhodovic/django-google-optimize#47] [General]
I'm looking for a way to add a/b testing for my Django project. I used to use optimise to make some front end changes, but want to make some backend changes now as well.

When I was looking through the documentation, I saw that this project is using the cookies that optimize sets. I was wondering how this works if it's the first request. 

E.g. in my case, I have a landing page that's served through Django. I want to try out different features on this landing page. But as I understand the cookie will be set after the LP is fully rendered & optimise js SDK is initialised. 

Will this library not work on the Landing page but only on subsequent pages? Do I understand this correctly?

**Comment by adinhodovic** [comment]
> I'm looking for a way to add a/b testing for my Django project. I used to use optimise to make some front end changes, but want to make some backend changes now as well.
> 
> When I was looking through the documentation, I saw that this project is using the cookies that optimize sets. I was wondering how this works if it's the first request.
> 
> E.g. in my case, I have a landing page that's served through Django. I want to try out different features on this landing page. But as I understand the cookie will be set after the LP is fully rendered & optimise js SDK is initialised.
> 
> Will this library not work on the Landing page but only on subsequent pages? Do I understand this correctly?

Yeah, there's an open issues. I'll probably get back to this soon in the future.

**Comment by etticat** [comment]
Ohh sorry. Didn't see that one. Thanks

I had a look but don't see a way to get around this. Do you have any ideas if there are ways to solve this? 

e.g. instead of optimising setting the a/b group, to let the backend define the group (by random)?
